src/lib/schemas/deep-research.ts

```diff
@@ -10,10 +10,18 @@ export const DeepResearchJobInputSchema = z.object({
 
 export type DeepResearchJobInput = z.infer<typeof DeepResearchJobInputSchema>;
 
+// Since we can't define a circular type directly in Zod, we use a more generic structure
+// that can represent our RerankResult<SearchGroups> type
+const RerankResultItemSchema = z.object({
+  group: z.string(),
+  item: z.any(),
```

Comment:
No `any` please. Avoid recursive types but if needed here's the explanation on howw to do it:

[Recursive objects](https://zod.dev/api?id=recursive-objects)

To define a self-referential type, use a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) on the key. This lets JavaScript resolve the cyclical schema at runtime.

const Category = z.object({
  name: z.string(),
  get subcategories(){
    return z.array(Category)
  }
});
 
type Category = z.infer<typeof Category>;
// { name: string; subcategories: Category[] }